[BUG] test_parquet_testing_valid_files for parquet-testing/data/null_list.parquet falls back

**Describe the bug**
```
FAILED ../../../../integration_tests/src/main/python/parquet_testing_test.py::test_parquet_testing_valid_files[confs0-/home/gshegalov/gits/NVIDIA/spark-rapids.26.02/thirdparty/parquet-testing/data/null_list.parquet][DATAGEN_SEED=1769835297, TZ=UTC, INJECT_OOM] - pyspark.errors.exceptions.captured.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ColumnarToRowExec
ColumnarToRow
+- FileScan parquet [emptylist#369] Batched: true, DataFilters: [], Format: Parquet, Location: InMemoryFileIndex(1 paths)[file:/home/gshegalov/gits/NVIDIA/spark-rapids.26.02/thirdparty/parquet..., PartitionFilters: [], PushedFilters: [], ReadSchema: struct<emptylist:array<void>>
FAILED ../../../../integration_tests/src/main/python/parquet_testing_test.py::test_parquet_testing_valid_files[confs1-/home/gshegalov/gits/NVIDIA/spark-rapids.26.02/thirdparty/parquet-testing/data/null_list.parquet][DATAGEN_SEED=1769835297, TZ=UTC] - pyspark.errors.exceptions.captured.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ColumnarToRowExec
ColumnarToRow
+- FileScan parquet [emptylist#901] Batched: true, DataFilters: [], Format: Parquet, Location: InMemoryFileIndex(1 paths)[file:/home/gshegalov/gits/NVIDIA/spark-rapids.26.02/thirdparty/parquet..., PartitionFilters: [], PushedFilters: [], ReadSchema: struct<emptylist:array<void>>
```
**Steps/Code to reproduce bug**

```
$ ~/gits/NVIDIA/spark-rapids.26.02$ PYSP_TST_spark_rapids_sql_explain=ALL TEST_PARALLEL=0 JAVA_HOME=/usr/lib/jvm/java-17-openjdk-amd64 SPARK_HOME=~/dist/spark-4.1.1-bin-hadoop3  ./integration_tests/run_pyspark_from_build.sh -k test_parquet_testing_valid_files -s |& tee test.log
```
```
pyspark.errors.exceptions.captured.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ColumnarToRowExec
ColumnarToRow
```

```
!Exec <FileSourceScanExec> cannot run on GPU because unsupported data types ArrayType(NullType,true) [emptylist] in read for Parquet
```

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**
Add any other context about the problem here.

--- a/integration_tests/src/main/python/parquet_testing_test.py
+++ b/integration_tests/src/main/python/parquet_testing_test.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2023-2025, NVIDIA CORPORATION.
+# Copyright (c) 2023-2026, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -22,7 +22,7 @@ from data_gen import copy_and_update, non_utc_allow
 from marks import allow_non_gpu
 from pathlib import Path
 import pytest
-from spark_session import is_before_spark_330, is_spark_350_or_later
+from spark_session import is_before_spark_330, is_spark_350_or_later, is_spark_411_or_later
 import warnings
 
 _rebase_confs = {
@@ -72,6 +72,12 @@ if is_spark_350_or_later():
 else:
     _error_files["lz4_raw_compressed.parquet"] = "Exception"
     _error_files["lz4_raw_compressed_larger.parquet"] = "Exception"
+
+if is_spark_411_or_later():
+    # XFAIL null_list.parquet for Spark 4.1.0+ due to https://github.com/NVIDIA/spark-rapids/issues/14242
+    # Before Spark 4.1.0, infers array<int>  type for the null_list.parquet,
+    # After Spark 4.1.0+, infers array<void> type for the null_list.parquet
+    _xfail_files["null_list.parquet"] = "https://github.com/NVIDIA/spark-rapids/issues/14242"
 
 def hdfs_glob(path_str, pattern):
     """

